### PR TITLE
Mustafa submission component - global and question specific docs

### PR DESCRIPTION
**Pull request by id-mustafa**
I finally have my feature in somewhat of a pushable state, however, it hasn't been completed just yet. Here is what I have gotten done so far:

- The question page calls on Nick's API to retrieve the questions from the backend.
- The questions page also has clickable questions that have a state set so it can be passed into my widget.
- The submission widget component has a submission and docs tab that takes in props from the question component and 
   currently only updates to have the global docs (question specific documents coming soon).
- Add the tab to switch between global and question specific docs. 

Here is what's next for me:
- Need to display the markdown in a separate tab.  (I've been stuck here).

- Add the code-editor IDE so that people can drop some code in here with syntax highlighting. - Task Difficulty: Medium

- Begin integrating Judge0 and try to build an API request to export code to the backend and from the backend to Judge0 and all the way back to the frontend. - Task Difficulty: Hard

### CODE REVIEW

**Comment by ItIsAndrewL** [comment]
@id-mustafa please remove library files from the git history and provide a way to install them in the instillation steps.

**Comment by ItIsAndrewL** [comment]
> @id-mustafa please remove library files from the git history and provide a way to install them in the instillation steps. Diffs this large crash github for me.

I think I see the issue. Looks like you installed `prisimjs` in a new top-level `node_modules` folder instead of the `frontend/node_modules` folder.

**Comment by ItIsAndrewL** [comment]
image

Error I'm receiving on startup on your branch.

**Comment by ItIsAndrewL** [comment]
*Note: Must squash and merge

File: frontend/src/components/SubmissionWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      ? `/docs/global/${docTitle}`
+      : `/docs/question/${question.num}/${docTitle}`;
+  };
+
+  return (
+    <section className="w-full max-w-lg h-[95vh] mx-auto bg-white rounded-lg shadow-md p-4 mt-1 mb-2 lg:ml-auto lg:mr-4 relative">
+      <div className="flex border-b">
+        <button
+          onClick={() => handleTabSwitch("submission")}
+          className={`w-1/2 py-2 text-center ${
```
Comment (ItIsAndrewL) [inline frontend/src/components/SubmissionWidget.tsx:36]: This is a super cool way to handle different styles on a condition!

File: frontend/src/pages/Question.tsx [hunk 2]
```diff
@@ -1,11 +1,65 @@
...
+      })
+      .catch((error) => {
+        console.error("Error:", error);
+      });
+  }, []);
+
+  // State stores the selected question.
+  const handleQuestionClick = (questionNum: number) => {
+    const question = questions?.find((q) => q.num - 1 === questionNum) || null;
+    setSelectedQuestion(question);
```
Comment (ItIsAndrewL) [inline frontend/src/pages/Question.tsx:41]: You _might_ be able to take advantage of the react routers sub routes to do this as well. I'm unsure if it'll work though so just a suggestion.

**Review by ItIsAndrewL — COMMENTED** [review]
Hey yall! Great work so far. I did a little code review on my phone and these are mostly just suggestions/questions or little nits to help yall keep working as it doesn't look completely finished yet. But keep going it's coming together 💪💪

**Comment by ItIsAndrewL** [comment]
Was able to give it a shot and looks good! Just getting some 404 errors when I attempt to open the documentation currently. I think once we fix that and you look at my comments above we can probably go ahead and merge this in. We should probably make note for the following things though:

- Refactor the docs pages to instead pull from the  `/questions` API route that we run at the beginning of the testing session.
- Add some error handling
- Update formatting to support markdown in the questions
- Add styling for the docs list
- Update styling to use the styles/colors we already have so we can have a more cohesive design

Also I'm unsure on how you're displaying the submission widget and if that knows which question we are on, since we will need that info for running the submission route (this can be a separate ticket if that's the case)

I think moving forward we should prefer marking things as TODOs and putting them as tickets for us to split up rather than just having you two work until things are perfect.

File: backend/main.py [hunk 3]
```diff
@@ -23,19 +23,19 @@
...
-        # ! Insert Tags Here
         count.openapi_tags,
         team.openapi_tags,
         auth.openapi_tags,
         question.openapi_tags,
+        docs.openapi_tags,
     ],
 )
 
-# Using GZip middleware is an internet standard for compressing HTML responses over the network
```
[the remark this answers is not in the document]
Comment (id-mustafa) [reply backend/main.py:34]: Removed by accident lol!

File: README.md [hunk 4]
```diff
@@ -51,6 +51,12 @@ Now move to App.tsx. Import the page from the component you had just created. an
 <Route path="[PATH THAT SHOWS IN THE URL]" element ={<[YOUR PAGE] />} />
 ```
 
+Due to frontend dependencies with viewing markdown, make sure you run:
+
+```bash
+npm install react-markdown remark-gfm github-markdown-css prismjs
```
[the remark this answers is not in the document]
Comment (id-mustafa) [reply README.md:57]: Resolved!

File: backend/api/docs.py
[the remark this answers is not in the document]
Comment (id-mustafa) [reply backend/api/docs.py:1]: Resolved! Used the originial route.

File: backend/main.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (id-mustafa) [reply backend/main.py:34]: Added it back ;)

File: frontend/src/components/MarkdownViewer.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    docType === "global"
+      ? `/docs/global/${docTitle}.md`
+      : `/docs/question/${questionNum}/doc_${docTitle}.md`;
+
+  const [content, setContent] = useState<string>("");
+
+  useEffect(() => {
+    const fetchMarkdown = async () => {
+      try {
+        const res = await fetch(finalUrl);
```
[the remark this answers is not in the document]
Comment (id-mustafa) [reply frontend/src/components/MarkdownViewer.tsx]: I switched the implementation over to use that route.

File: frontend/src/components/SubmissionWidget.tsx (hunk 1 above)
Comment (id-mustafa) [reply frontend/src/components/SubmissionWidget.tsx:41]: Thanks dude!

File: frontend/src/pages/Question.tsx (hunk 2 above)
Comment (id-mustafa) [reply frontend/src/pages/Question.tsx:41]: I think this was Matthew's change. I'll leave this open for him. I had a different way of doing it.

**Comment by id-mustafa** [comment]
Refer to my other PR. I believe that one gets rid of the 404s and the errors you're facing and I agree with you on the styling I can add it to the backlog. The markdown renders fine now but you'll see the changes reflected in the other branch: mustafa-submission-component-2.

**Pull request closed without merging**